staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/unstructured/unstructured.go

```
@@ -226,8 +233,10 @@ func getNestedInt64Pointer(obj map[string]interface{}, fields ...string) *int64
 	}
 }
 
-func getNestedSlice(obj map[string]interface{}, fields ...string) []string {
-	if m, ok := getNestedField(obj, fields...).([]interface{}); ok {
+// NestedStringSlice returns the []string value of a nested field.
+// Returns nil if value is not found or is not a []interface{}.
+func NestedStringSlice(obj map[string]interface{}, fields ...string) []string {
+	if m, ok := NestedField(obj, fields...).([]interface{}); ok {
 		strSlice := make([]string, 0, len(m))
 		for _, v := range m {
 			if str, ok := v.(string); ok {
```

I would prefer to return an error if the cast fails.